### ISSUE TITLE
Allow single object as parameter to query a set/list 

### ISSUE THREAD

**Issue by hannes-angst** [boostchicken/spring-data-dynamodb#33]
My entity contains a field of type Set<String> countries (stringSet). I want to query against it with one country.

## Expected Behavior

`Iterable<ProductEntity> findAllByCountriesContainsOrCountriesContains(String country)` returns a list of products, that contains `country` their `countries` fields.

## Actual Behavior

```
java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.Collection (java.lang.String and java.util.Collection are in module java.base of loader 'bootstrap')
	at com.amazonaws.services.dynamodbv2.datamodeling.StandardTypeConverters$Vector$ToSet$1.convert(StandardTypeConverters.java:449) ~[aws-java-sdk-dynamodb-1.11.415.jar:na]
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter$DelegateConverter.convert(DynamoDBTypeConverter.java:104) ~[aws-java-sdk-dynamodb-1.11.415.jar:na]
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter$NullSafeConverter.convert(DynamoDBTypeConverter.java:123) ~[aws-java-sdk-dynamodb-1.11.415.jar:na]
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter$ExtendedConverter.convert(DynamoDBTypeConverter.java:83) ~[aws-java-sdk-dynamodb-1.11.415.jar:na]
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperFieldModel.convert(DynamoDBMapperFieldModel.java:138) ~[aws-java-sdk-dynamodb-1.11.415.jar:na]
	at org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCriteria.getPropertyAttributeValue(AbstractDynamoDBQueryCriteria.java:557) ~[spring-data-dynamodb-5.2.4.jar:5.2.4]
	at org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCriteria.createSingleValueCondition(AbstractDynamoDBQueryCriteria.java:706) ~[spring-data-dynamodb-5.2.4.jar:5.2.4]
	at org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCriteria.withSingleValueCriteria(AbstractDynamoDBQueryCriteria.java:485) ~[spring-data-dynamodb-5.2.4.jar:5.2.4]
	at org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCreator.addCriteria(AbstractDynamoDBQueryCreator.java:151) ~[spring-data-dynamodb-5.2.4.jar:5.2.4]
	at org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCreator.create(AbstractDynamoDBQueryCreator.java:123) ~[spring-data-dynamodb-5.2.4.jar:5.2.4]
	at org.socialsignin.spring.data.dynamodb.repository.query.AbstractDynamoDBQueryCreator.create(AbstractDynamoDBQueryCreator.java:49) ~[spring-data-dynamodb-5.2.4.jar:5.2.4]
```	
## Specifications
  
  - Spring Data DynamoDB Version: 5.2.4 (2.2)
  - Spring Data Version: 2.3.1.RELEASE
  - AWS SDK Version: 1.11.415
  - Java Version: 11.0.2 - OpenJDK 64-Bit Server VM 11.0.2+9
  - Platform Details: Mac OS X 10.15.4

**Comment by boostchicken** [comment]
@hannes-angst great work, if you have any interest, I am looking for other contributors to the project.

**Comment by boostchicken** [comment]
@hannes-angst released in 5.2.5-SNAPSHOT on ossrh
